packages/flutter/lib/src/cupertino/sliding_segmented_control.dart

```diff
@@ -649,6 +693,22 @@ class _SegmentedControlState<T extends Object> extends State<CupertinoSlidingSeg
         );
       }
 
+      bool enabled = true;
+      if (widget.setEnabled != null && widget.setEnabled!.containsKey(entry.key)) {
+        enabled = widget.setEnabled![entry.key]!;
+      }
```

Comment:
`final bool enabled = widget.setEnabled?.[entry.key] ?? true`.

Also you can use a set to store disabled segments.

Reply:
Updated!